x-pack/plugins/search_playground/server/utils/stream_factory.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/core/server';
+
+import { PassThrough } from 'stream';
+
+import type { ResponseHeaders } from '@kbn/core-http-server';
+import { repeat } from 'lodash';
+
+const DELIMITER = `
+`;
+
+export interface StreamResponseWithHeaders {
+  body: PassThrough;
+  headers?: ResponseHeaders;
+}
+
+export interface StreamFactoryReturnType {
+  DELIMITER: string;
+  end: () => void;
+  push: (d: string) => void;
+  responseWithHeaders: StreamResponseWithHeaders;
+}
+
+export function streamFactory(logger: Logger, flushFix: boolean = true): StreamFactoryReturnType {
```

Comment:
What did we change here to need to copy this for ourselves?

Reply:
added comments on top of the file with the changes. thanks!